src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -292,8 +319,131 @@ internal Cci.IFieldReference CreateDataField(ImmutableArray<byte> data, ushort a
                     _ => hex
                 };
 
-                return new MappedField(name, this, type, key.Data);
-            });
+                return new MappedField(name, @this, type, data);
+            },
+            this);
+        }
+
+        internal DataSectionStringType? GetOrCreateDataSectionStringLiteralType(
```

Comment:
>GetOrCreateDataSectionStringLiteralType [](http://example.com/codeflow?start=40&length=39)

It doesn't look like anyone is interested in the type. The only consumer is interested in a field. I think we need to adjust this API accordingly.